micropython_ota/micropython_ota.py

```diff
@@ -0,0 +1,45 @@
+import machine
+from urequests import urequests
+import os
+
+
+def check_version(host, project):
+    try:
+        if 'version' in os.listdir():
+            with open('version', 'r') as current_version_file:
+                current_version = current_version_file.readline().strip()
+        else:
+            current_version = ''
+        remote_version_response = urequests.get(f'{host}/{project}_version')
+        if remote_version_response.status_code != 200:
+            print(f'Remote version file {host}/{project}_version not found')
+            return False, current_version
+        remote_version = remote_version_response.text.strip()
+        return current_version != remote_version, remote_version
+    except Exception as ex:
+        print(f'Something went wrong: {ex}')
+        return False, current_version
+
+
+def ota_update(host, project, filename):
+    try:
+        version_changed, remote_version = check_version(host, project)
+        if version_changed:
+            source_file_response = urequests.get(f'{host}/{project}_{remote_version}_{filename}')
```

Comment:
Instead of prefixing the source file with the project name, using a context path would be more suitable.